File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {        
        Stack<Integer> s = new Stack<Integer>();
        
        for (int elem : arr) {
            if (s.empty()) s.push(elem);
            else {
                if (s.peek() != elem) s.push(elem);
            }
        }
        
        List<Integer> answer = new ArrayList<>();
        while (!s.empty()) {
            answer.add(s.pop());
        }
        Collections.reverse(answer);

        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <stack>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    
    stack<int> s;
    for (auto elem : arr) {
        if (s.empty() || s.top() != elem) s.push(elem);
    }
    
    while (!s.empty()) {
        answer.push_back(s.top());
        s.pop();
    }
    
    reverse(answer.begin(), answer.end());

    return answer;
}